src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java

```diff
@@ -38,20 +39,22 @@
 import org.jetbrains.annotations.NotNull;
 
 import java.util.ArrayList;
+import java.util.LinkedHashMap;
 import java.util.List;
+import java.util.Map;
 
 import static com.minecolonies.coremod.entity.ai.citizen.guard.GuardConstants.*;
 import static com.minecolonies.coremod.entity.ai.util.AIState.*;
 
-@SuppressWarnings("squid:MaximumInheritanceDepth")
```

Comment:
### Errors:
- Overly complex class AbstractEntityAIGuard (cyclomatic complexity = 103) 
- Class 'AbstractEntityAIGuard' has too many dependencies (26 > 10)
- Class 'AbstractEntityAIGuard' does not override 'toString()' method
- Class 'AbstractEntityAIGuard' has too many transitive dependencies (669 > 35)